src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -48,11 +48,16 @@ public TxReceipt Find(Keccak hash)
             var receiptData = _database.Get(hash);
             if (receiptData != null)
             {
-                var receipt = Rlp.Decode<TxReceipt>(new Rlp(receiptData), RlpBehaviors.Storage);
-                receipt.TxHash = hash;
-                return receipt;
+                try
+                {
+                    var receipt = Rlp.Decode<TxReceipt>(new Rlp(receiptData), RlpBehaviors.Storage);
+                    receipt.TxHash = hash;
+                    return receipt;
+                }
+                // During synchronization TxReceipt was inserted with RlpBehaviors.None but this transaction wasn't processed
+                // Will be fixed with future block receipts refactoring
+                catch (RlpException) { }
```

Comment:
shall we fill all the data? otherwise some important data may be missing from response

Reply:
Sorry don't understand what you mean

Reply:
here we return Receipt but if we retrieve it for RPC calls then some properties will not be populated

Reply:
I say that we should go to eth.getTransactionReceipt and whenever we read a limited data for receipt then we should fill in from transaction and block. if this is already done then ignore

Reply:
with Storage flag it is already filled